src/pybind/mgr/diskprediction/module.py

```diff
@@ -0,0 +1,346 @@
+        {
+            'cmd': 'diskprediction self-test',
+            'desc': 'Prints hello world to mgr.x.log',
+            'perm': 'r'
+        },
+        {
+            'cmd': 'diskprediction status',
+            'desc': 'Check diskprediction status',
+            'perm': 'r'
+        }
+    ]
+
+    def __init__(self, *args, **kwargs):
+        super(Module, self).__init__(*args, **kwargs)
+        self.status = DP_MGR_STAT_DISABLED
+        self.event = Event()
+        self.run = True
+        self._tasks = []
+        self._activate_cloud = False
+        self._prediction_result = {}
+
+    @property
+    def config_keys(self):
+        return dict((o['name'], o.get('default', None)) for o in self.OPTIONS)
+
+    def set_config_option(self, option, value):
+        if option not in self.config_keys.keys():
+            raise RuntimeError('{0} is a unknown configuration '
+                               'option'.format(option))
+
+        if option in ['diskprediction_port',
+                      'diskprediction_upload_metrics_interval',
+                      'diskprediction_upload_smart_interval',
+                      'diskprediction_retrieve_prediction_interval']:
+            try:
+                value = int(value)
+            except (ValueError, TypeError):
+                raise RuntimeError('invalid {0} configured. Please specify '
+                                   'a valid integer'.format(option))
+
+        self.log.debug('Setting in-memory config option %s to: %s', option,
+                       value)
+        self.config[option] = value
+        return True
+
+    def get_configuration(self, key):
+        return self.get_config(key, self.config_keys[key])
+
+    def _config_show(self, inbuf, cmd):
+        self.show_module_config()
+        return 0, json.dumps(self.config, indent=4), ''
+
+    def _config_mode(self, inbuf, cmd):
+        str_mode = cmd.get('mode', 'cloud')
+        if str_mode.lower() not in ['cloud', 'onpremise', 'local']:
+            return 0, 'invalid configuration, enable=[cloud|onpremise|local]', ''
+        try:
+            self.set_config('diskprediction_config_mode', str_mode)
+            return (0,
+                    'success to config disk prediction mode: %s'
+                    % str_mode.lower(), 0)
+        except Exception as e:
+            return 0, str(e), ''
+
+    def _set_smart_relied_device_health(self, inbuf, cmd):
+        str_relied = cmd.get('relied', 'false')
+        if str_relied.lower() not in ['false', 'true']:
+            return 0, 'invalid configuration, enable=[true|false]', ''
+        try:
+            self.set_config('diskprediction_smart_relied_device_health', str_relied)
+            return (0,
+                    'success to config device data %s on the device plugin'
+                    % 'not relied' if str_relied.lower() == 'false' else 'relied', 0)
+        except Exception as e:
+            return 0, 'command failed, %s' % str(e), ''
```

Comment:
Do not return 0 on errors, and I'm not sure it's necessary to catch an exception here at all -- under what circumstances would set_config fail?